### PR TITLE
[Keymap] Adds ymdk/ymd09 andys8 layout

### PR DESCRIPTION
**Pull request by andys8**
## Description

Custom layout for macropad. It shows the usage of macros with unicode, and other layouts (ISO-DE) with unicode, emojis, and git commands.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ymdk/ymd09/keymaps/andys8/config.h [hunk 1]
```diff
@@ -0,0 +1 @@
+#define UNICODE_SELECTED_MODES UC_LNX
```
Comment (drashna) [inline keyboards/ymdk/ymd09/keymaps/andys8/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

And this needs `#pragma once` before the define.

File: keyboards/ymdk/ymd09/keymaps/andys8/keymap.c [hunk 2]
```diff
@@ -0,0 +1,81 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/ymdk/ymd09/keymaps/andys8/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/ymdk/ymd09/keymaps/andys8/keymap.c [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            break;
+        case E_SHRUG:
+            if (record->event.pressed) {
+                send_unicode_string("¯\\_(ツ)_/¯");
+            }
+            break;
+        case GIT_STASH:
+            if (record->event.pressed) {
+                send_string("git stash ");
+                tap_code(DE_MINS);
```
Comment (drashna) [inline keyboards/ymdk/ymd09/keymaps/andys8/keymap.c:40]: If you include `sendstring_german.h`, you should be able to use a single function.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/ymdk/ymd09/keymaps/andys8/keymap.c (hunk 3 above)
Comment (andys8) [reply keyboards/ymdk/ymd09/keymaps/andys8/keymap.c]: Thanks for this great hint. Makes everything a lot better.

File: keyboards/ymdk/ymd09/keymaps/andys8/config.h (hunk 1 above)
Comment (andys8) [reply keyboards/ymdk/ymd09/keymaps/andys8/config.h:18]: Added license and pragma

File: keyboards/ymdk/ymd09/keymaps/andys8/keymap.c (hunk 2 above)
Comment (andys8) [reply keyboards/ymdk/ymd09/keymaps/andys8/keymap.c:17]: Added license

**Review by drashna — APPROVED** [review]

**Comment by andys8** [comment]
@drashna I noticed that sending unicode strings is working in less places, than it initially did.
E.g. in the chrome address bar it's still working, but not anymore in the terminal.

Do you have an idea if this is related to the operating system, updates in qmk `master` branch, or due to changes in this PR?

## Chrome working

image


## Terminal

image

## Qutebrowser

image

File: keyboards/ymdk/ymd09/keymaps/andys8/keymap.c [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+            }
+            break;
+        case E_SHRUG:
+            if (record->event.pressed) {
+                send_unicode_string("¯\\_(ツ)_/¯");
+            }
+            break;
+        case GIT_STASH:
+            if (record->event.pressed) {
+                send_string("git stash"SS_TAP(X_ENTER));
```
Comment (fauxpark) [inline keyboards/ymdk/ymd09/keymaps/andys8/keymap.c:55]:
```suggestion
                SEND_STRING("git stash\n");
```

File: keyboards/ymdk/ymd09/keymaps/andys8/keymap.c [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+            }
+            break;
+        case GIT_STASH_POP:
+            if (record->event.pressed) {
+                send_string("git stash pop"SS_TAP(X_ENTER));
```
Comment (fauxpark) [inline keyboards/ymdk/ymd09/keymaps/andys8/keymap.c:60]:
```suggestion
                SEND_STRING("git stash pop\n");
```

File: keyboards/ymdk/ymd09/keymaps/andys8/keymap.c [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+            }
+            break;
+        case GIT_COMMIT:
+            if (record->event.pressed) {
+                send_string("git add -A && git commit -a"SS_TAP(X_ENTER));
```
Comment (fauxpark) [inline keyboards/ymdk/ymd09/keymaps/andys8/keymap.c:65]:
```suggestion
                SEND_STRING("git add -A && git commit -a\n");
```

File: keyboards/ymdk/ymd09/keymaps/andys8/keymap.c [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+            }
+            break;
+        case COPY_PASTE:
+            if (record->event.pressed) {
+                send_string(SS_LCTL("c"));
+            } else {
+                send_string(SS_LCTL("v"));
```
Comment (fauxpark) [inline keyboards/ymdk/ymd09/keymaps/andys8/keymap.c:72]:
```suggestion
                tap_code16(C(KC_C));
            } else {
                tap_code16(C(KC_V));
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by drashna** [comment]
Unicode or the like is a very tricky subject, because it is NOT handled consistently. Not between OS's, and sometimes, not even between programs. I don't really have a good answer for you, aside from tinkering and find what works.   

That said, there is the unicode feature in QMK which may work better.

File: keyboards/ymdk/ymd09/keymaps/andys8/keymap.c (hunk 4 above)
Comment (andys8) [reply keyboards/ymdk/ymd09/keymaps/andys8/keymap.c:31]: Done. I'm actually surprised that `\n` is working. That looks easy.

File: keyboards/ymdk/ymd09/keymaps/andys8/keymap.c (hunk 5 above)
Comment (andys8) [reply keyboards/ymdk/ymd09/keymaps/andys8/keymap.c:36]: ✅

File: keyboards/ymdk/ymd09/keymaps/andys8/keymap.c (hunk 6 above)
Comment (andys8) [reply keyboards/ymdk/ymd09/keymaps/andys8/keymap.c:41]: ✅

File: keyboards/ymdk/ymd09/keymaps/andys8/keymap.c (hunk 7 above)
Comment (andys8) [reply keyboards/ymdk/ymd09/keymaps/andys8/keymap.c:48]: This one is also an improvement. Thanks ✅

**Comment by andys8** [comment]
I dropped unicode handling all together, because it wasn't reliable with different systems and applications. I'm going to handle this in the system. I added the suggested code improvements, thanks.

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**